mat-process-api/V1/Controllers/ProcessDataController.cs

```diff
@@ -54,21 +58,40 @@ public IActionResult GetProcessData(string processRef)
         /// If processData needs to be update, it's instead replaced with a full new one.
         /// Returning flat out 200 on successful update
         /// </summary>
-        /// <param name="processDataJSON"></param>
+        /// <param name="updateRequest"></param>
         /// <returns></returns>
-        [HttpPatch]
+        [HttpPut]
         [Produces("application/json")]
         [ProducesResponseType(StatusCodes.Status200OK)]
-        public IActionResult UpdateExistingProcessDocument()
+        public IActionResult UpdateExistingProcessDocument([FromBody]UpdateProcessDataRequest updateRequest)
         {
-            return Ok();
+            var isRequestValid = _updateValidator.Validate(updateRequest);
+            if (isRequestValid.IsValid)
+            {
+
+                try
+                {
+                    _logger.LogInformation($"Update ProcessData request for process ID {updateRequest.processDataToUpdate.Id}");
+                    var result = _processDataUsecase.ExecuteUpdate(updateRequest);
+                    return Ok(result);
+                }
+                catch (DocumentNotFound ex)
+                {
+                    return StatusCode(200, $"Document with reference {updateRequest.processDataToUpdate.Id} was not found in the database." +
```

Comment:
I think it should be this, stated reasoning next to the tests: 404 | Not found - the resource is not found.
Also, according to the playbook:
"PATCH requests are usually not robust against non-existence of resource instances", which implies that we should use an error response rather than success one.